tests/HTMLPurifierTest.hack

```diff
@@ -466,4 +466,83 @@ class HTMLPurifierTest extends HackTest {
 		expect($clean_html)->toEqual($expected_html);
 		echo "finished.\n\n";
 	}
+
+	public function testRel(): void {
+		echo "\nrunning testRel()...";
+		$policy = new HTMLPurifier\HTMLPurifier_Policy(
+			dict[
+				'b' => vec[],
+				'ul' => vec[],
+				'li' => vec[],
+				'ol' => vec[],
+				'h2' => vec[],
+				'h4' => vec[],
+				'br' => vec[],
+				'div' => vec[],
+				'strong' => vec[],
+				'del' => vec[],
+				'em' => vec[],
+				'pre' => vec[],
+				'code' => vec[],
+				'table' => vec[],
+				'tbody' => vec[],
+				'td' => vec[],
+				'th' => vec[],
+				'thead' => vec[],
+				'tr' => vec[],
+				'a' => vec['id', 'name', 'href', 'target', 'rel'],
+				'h3' => vec['class'],
+				'p' => vec['class'],
+				'aside' => vec['class'],
+				'img' => vec['src', 'alt', 'class', 'width', 'height', 'srcset', 'sizes'],
+			],
+		);
+		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
+		$dirty_html = '<p><a rel="ab" name="bobcat"></a></p>';
+		$clean_html = $purifier->purify($dirty_html);
+		$expected_html = '<p><a name="bobcat"></a></p>';
+		expect($clean_html)->toEqual($expected_html);
+		echo "finished.\n\n";
+	}
+
+
+	public function testAtagTargetAttribute(): void {
+		echo "\nrunning testAtagTargetAttribute()...";
+		$policy = new HTMLPurifier\HTMLPurifier_Policy(
+			dict[
+				'b' => vec[],
+				'ul' => vec[],
+				'li' => vec[],
+				'ol' => vec[],
+				'h2' => vec[],
+				'h4' => vec[],
+				'br' => vec[],
+				'div' => vec[],
+				'strong' => vec[],
+				'del' => vec[],
+				'em' => vec[],
+				'pre' => vec[],
+				'code' => vec[],
+				'table' => vec[],
+				'tbody' => vec[],
+				'td' => vec[],
+				'th' => vec[],
+				'thead' => vec[],
+				'tr' => vec[],
+				'a' => vec['id', 'name', 'href', 'target', 'rel'],
+				'h3' => vec['class'],
+				'p' => vec['class'],
+				'aside' => vec['class'],
+				'img' => vec['src', 'alt', 'class', 'width', 'height', 'srcset', 'sizes'],
+			],
+		);
+		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
+		$dirty_html = '<p><a name="bobcat" target="_blank" rel="noopener noreferrer"></a></p>';
```

Comment:
Can we have a test where the `rel` is not supplied and make sure that it is added correctly?

Reply:
I think there is a confusion about `rel`.
When `target="_blank"` exist but `rel` doesn't, html sanitizer will not add `rel="noopener noreferrer"` attribute to `a` tag.

Reply:
Hmm, I think it probably should? (cc @id4ho )

Reply:
From the thread:

> I think the weirdness on the lib was handling link with target attribute. If the link have target=_blank it will add that rel=\"noreferrer noopener\" to the output result even if the policy weren't set to allow rel.

Reply:
If we check on the four diffs, `target="_blank" rel="noopener noreferrer"` exists in the original dirty html. I didn't find any occurrence when `target="_blank"` exists in original dirty html and `rel="noopener noreferrer` is added in php clean html.

Reply:
However the correct behavior for this library would be to add `rel="noopener noreferrer"` if it's not set and `target="_blank:"` so let's add a test to make sure it does, and fix it if it doesn't (looks like it might only need a config change).  This is an important security feature so we should make sure it's covered.

Reply:
So even if it's not present in the existing html that we are sanitizing that is what the php service would do per fikrie's test in that thread. So I think we should add that test, because it does ensure we don't get out of step with the current behavior of the php service and as O said, it's also the proper behavior from a security pov.